Can disable logging on demand

LasseRafn opened the pull request:
I was working on a way to disable on `::create` however it proved to be harder than expected.

Please modify the code if you have any way of improving it and making it visually prettier to look at.

I've added 3 tests to support this.

## Usage

```php
$user = User::find(1);

// Example 1:
$user->withoutActivityLogging()->update(['name' => 'Testing']);

// Example 2:
$user->withoutActivityLogging();
$user->name = 'Testing';
$user->save();
```

### Re-enabling logging for same model

Should you want to enable logging again, I've added another method:

```php
$user = User::find(1);

// Disable
$user->withoutActivityLogging()->update(['name' => 'Not logged']);

// Enable
$user->withActivityLogging(); // with, not without
$user->name = 'Logged';
$user->save();
```

LasseRafn commented:
#226

freekmurze commented on src/Traits/LogsActivity.php:
```diff
@@ -14,6 +13,8 @@
 {
     use DetectsChanges;
 
+    private $withoutActivityLogging = false;
```
In my mind it's easier to reason about these kind of boolean, where they are named in a positive.

Rename this to `enableLoggingModelsEvents` (or something better) with a default value of `true`

freekmurze replied:
Also make it protected

freekmurze commented on tests/LogsActivityTest.php:
```diff
@@ -9,206 +9,247 @@
 {
-    /** @var \Spatie\Activitylog\Test\Article|\Spatie\Activitylog\Traits\LogsActivity */
-    protected $article;
+	/** @var \Spatie\Activitylog\Test\Article|\Spatie\Activitylog\Traits\LogsActivity */
```
There seems to be something off with this diff. I can't easily see what changes you did make in regards to the PR. Could you only add lines and not modify existing ones that have nothing to do with this PR?

LasseRafn replied:
Sorry, seems that my IDE decided to screw with the style 👎 I'll fix this

LasseRafn replied:
This has been solved. I've added a Spatie-specific PhpStorm style file for future PRs 👍

damosse31 commented:
Any news about this PR guys ?

damosse31 commented:
@freekmurze @LasseRafn is this improve dead ?

LasseRafn commented:
I was working on a way to disable on `::create` however it proved to be harder than expected.

Please modify the code if you have any way of improving it and making it visually prettier to look at.

I've added 3 tests to support this.

## Usage

```php
$user = User::find(1);

// Example 1:
$user->withoutActivityLogging()->update(['name' => 'Testing']);

// Example 2:
$user->withoutActivityLogging();
$user->name = 'Testing';
$user->save();
```

### Re-enabling logging for same model

Should you want to enable logging again, I've added another method:

```php
$user = User::find(1);

// Disable
$user->disableLogging()->update(['name' => 'Not logged']);

// Enable
$user->enableLogging();
$user->name = 'Logged';
$user->save();
```

LasseRafn commented:
@freekmurze docs.: https://github.com/spatie/docs.spatie.be/pull/186

freekmurze commented:
Thanks!
